### PR TITLE
infrastructure: Upgrade VM to Ubuntu 20.04, misc fixes and clean-up

### PR DESCRIPTION
**Pull request by CajuM**

### CODE REVIEW

**Comment by CajuM** [comment]
I should note that tests currently don't work for some homeworks.

File: infrastructure/vm/playbooks-common/helper-pkgs.yml [hunk 1]
```diff
@@ -1,22 +1,13 @@
 - name: Install helper packages (driver-level)
   apt:
+    # open-vm-tools is necesary for exporting to VMware
```
Comment (edi33416) [inline infrastructure/vm/playbooks-common/helper-pkgs.yml]:
```suggestion
    # open-vm-tools are necessary for exporting to VMware
```

**Review by edi33416 — CHANGES_REQUESTED** [review]
LGTM

Please also get in touch with someone from the Homeworks team to double check.

Have you checked that the VM is working properly with the homeworks skeletons?

**Comment by CajuM** [comment]
> LGTM
> 
> Please also get in touch with someone from the Homeworks team to double check.
> 
> Have you checked that the VM is working properly with the homeworks skeletons?

I verified all the homework checkers. They all compile, but some throw errors when ran on the solution. At least in one case the solution was poorly written.

**Comment by CajuM** [comment]
* tema1: ok
* tema2: ok
* tema3: problem with solution
* tema4: ok
* team5: ok

File: infrastructure/vm/vagrantfile-so/publish.sh [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-vagrant cloud publish edi33416/so-ubuntu-18.04 1.1.0 virtualbox so-ubuntu-18-04.box -d "Ubuntu 18.04 based on bento/ubuntu-18.04 and provisioned with Ansible" --version-description "Latest stable version" --release --short-description "SO VM"
+vagrant cloud publish edi33416/so-ubuntu-20.04 1.1.0 virtualbox so-ubuntu-20-04.box -d "Ubuntu 20.04 based on bento/ubuntu-20.04 and provisioned with Ansible" --version-description "Latest stable version" --release --short-description "SO VM"
```
Comment (edi33416) [inline infrastructure/vm/vagrantfile-so/publish.sh]:
```suggestion
vagrant cloud publish edi33416/so-ubuntu-20.04 1.3.0 virtualbox so-ubuntu-20-04.box -d "Ubuntu 20.04 based on bento/ubuntu-20.04 and provisioned with Ansible" --version-description "Latest stable version" --release --short-description "SO VM"
```

Comment (edi33416) [reply infrastructure/vm/vagrantfile-so/publish.sh]: Please bump the version to 1.3.0, as there is 1.2.0 is the current [release](https://app.vagrantup.com/edi33416/boxes/so-ubuntu-18.04/versions/1.2.0)

File: infrastructure/vm/vagrantfile-so/publish.sh [hunk 3]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-vagrant cloud publish edi33416/so-ubuntu-18.04 1.1.0 virtualbox so-ubuntu-18-04.box -d "Ubuntu 18.04 based on bento/ubuntu-18.04 and provisioned with Ansible" --version-description "Latest stable version" --release --short-description "SO VM"
+vagrant cloud publish systems-cs-pub-ro/so 1.3.0 virtualbox so-ubuntu-20-04.box -d "Ubuntu 20.04 based on bento/ubuntu-20.04 and provisioned with Ansible" --version-description "Latest stable version" --release --short-description "SO VM"
```
Comment (edi33416) [inline infrastructure/vm/vagrantfile-so/publish.sh]:
```suggestion
vagrant cloud publish systems-cs-pub-ro/so 2.0.0 virtualbox so-ubuntu-20-04.box -d "Ubuntu 20.04 based on bento/ubuntu-20.04 and provisioned with Ansible" --version-description "Latest stable version" --release --short-description "SO VM"
```

**Review by edi33416 — APPROVED** [review]
Nice job!

Please make sure to cleanup the image before publishing it, to keep the size small.

Have a look at [this StackOverflow](https://stackoverflow.com/a/35733376) answer.

You could use [this ](https://gist.github.com/adrienbrault/3775253) cleanup script; probably you'll need to replace `#!/bin/sh` with `#!/bin/bash`

**Pull request merged**